Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,261 @@
+                {
+                    richestPerson = i;
+                }
+            }
+            return message = $"{workingBalances[richestPerson][0]} is the richest person. ${workingBalances[richestPerson][^1]}";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            string[][] workingBalances = ArrayifyBalances(peopleAndBalances);
+            int poorestPerson = 0;
+            string message = $"{workingBalances[poorestPerson][0]} is the poorest person. ${workingBalances[poorestPerson][^1]}";
+            if (workingBalances.Length == 1)
+            {
+                return message;
+            }
+            for (int i = 1; i < workingBalances.Length; i++)
+            {
+                if (float.Parse(workingBalances[i][^1]) < float.Parse(workingBalances[poorestPerson][^1]))
+                {
+                    poorestPerson = i;
+                }
+            }
+            return message = $"{workingBalances[poorestPerson][0]} is the poorest person. ${workingBalances[poorestPerson][^1]}";
+        }
+
+        /// <summary>
+        /// Takes an array of strings and returns a jagged array with each string on ", ".
+        /// </summary>
+        public static string[][] ArrayifyBalances(string[] peopleAndBalances)
+        {
+            string[][] parsedBalances = new string[peopleAndBalances.Length][];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] tempArr = peopleAndBalances[i].Split(", ", System.StringSplitOptions.None);
+                parsedBalances[i] = tempArr;
+            }
+
+            return parsedBalances;
+        }
+
+        /// <summary>
+        /// Checks if an array is NULL or empty and returns true if any of those conditions are true.
+        /// </summary>
+        public static bool IsArrayNullOrEmpty(string[] peopleAndBalances)
+        {
+            return peopleAndBalances == null || peopleAndBalances.Length == 0;
+        }
+
+        /// <summary>
+        /// Takes an array of strings and returns that array as floats.
+        /// </summary>
+        public static float[] StringArrToFloatArr(string[] balances)
+        {
+            int failLength = 0;
+            float[] floatArr = new float[balances.Length];
+            for (int i = 0; i < floatArr.Length; i++)
+            {
+                bool success = float.TryParse(balances[i], out float number);
+                if (success) floatArr[i - failLength] = number;
+                else failLength++;
+            }
+            if(failLength == 0) return floatArr;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)